src/youtube2zim/youtube.py

```diff
@@ -251,16 +258,20 @@ def retrieve_videos_for(videos_ids):
     return items
 
 
-def save_channel_branding(channels_dir, channel_id, save_banner=False):
+def save_channel_branding(channels_dir, channel_id, *, save_banner=False):
     """download, save and resize profile [and banner] of a channel"""
     channel_json = get_channel_json(channel_id)
 
     thumbnails = channel_json["snippet"]["thumbnails"]
+    thumnbail = None
     for quality in ("medium", "default"):  # high:800px, medium:240px, default:88px
         if quality in thumbnails.keys():
             thumnbail = thumbnails[quality]["url"]
             break
 
+    if not thumnbail:
```

Comment:
I'm not sure this is reproducing current behavior. If there was no `medium` nor `default` quality thumbnail (and I believe it's possible), then we were skipping it and continue.
With this, the scraper will halt.

Reply:
I agree that the `if not thumbnail` is not placed appropriately.
It should be within the `if not profile_path.exists`.

Reply:
I understand we can't not have a thumbnail with the current code. Was it the case? I think not but haven't checked. If it was possible not to have it, we should stick to it.

Reply:
Before this PR, when there was no profile image passed as argument and no thumbnail found, the code was crashing when we tried to stream the file with a hard to read error:

```
[youtube2zim::2023-10-20 08:09:11,288] ERROR:FAILED. An error occurred: Invalid URL 'None': No scheme supplied. Perhaps you meant https://None?
[youtube2zim::2023-10-20 08:09:11,288] ERROR:Invalid URL 'None': No scheme supplied. Perhaps you meant https://None?
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/youtube2zim/entrypoint.py", line 220, in main
    return scraper.run()
           ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/youtube2zim/scraper.py", line 321, in run
    self.download_authors_branding()
  File "/usr/local/lib/python3.11/site-packages/youtube2zim/scraper.py", line 762, in download_authors_branding
    save_channel_branding(self.channels_dir, channel_id, save_banner=False)
  File "/usr/local/lib/python3.11/site-packages/youtube2zim/youtube.py", line 279, in save_channel_branding
    stream_file(thumnbail, profile_path)
  File "/usr/local/lib/python3.11/site-packages/zimscraperlib/download.py", line 197, in stream_file
    resp = session.get(
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 575, in request
    prep = self.prepare_request(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 486, in prepare_request
    p.prepare(
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 368, in prepare
    self.prepare_url(url, params)
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 439, in prepare_url
    raise MissingSchema(
requests.exceptions.MissingSchema: Invalid URL 'None': No scheme supplied. Perhaps you meant https://None?
```

Now it is returning a "nicer" error message "thumnbail not found":
```
youtube2zim::2023-10-20 08:10:40,053] ERROR:FAILED. An error occurred: thumnbail not found
[youtube2zim::2023-10-20 08:10:40,053] ERROR:thumnbail not found
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/youtube2zim/entrypoint.py", line 220, in main
    return scraper.run()
           ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/youtube2zim/scraper.py", line 321, in run
    self.download_authors_branding()
  File "/usr/local/lib/python3.11/site-packages/youtube2zim/scraper.py", line 762, in download_authors_branding
    save_channel_branding(self.channels_dir, channel_id, save_banner=False)
  File "/usr/local/lib/python3.11/site-packages/youtube2zim/youtube.py", line 278, in save_channel_branding
    raise Exception("thumnbail not found")
Exception: thumnbail not found
```